src/legacy_hal/diagnostic.c

```diff
@@ -0,0 +1,1422 @@
+/***************************************************************************/
+/**
+ * @file
+ *
+ * @brief Diagnostic functions for the crash handler
+ *******************************************************************************
+ * # License
+ * <b>Copyright 2023 Silicon Laboratories Inc. www.silabs.com</b>
```

Comment:
Apply OpenThread license header or move to `third_party`?

Reply:
> Apply OpenThread license header or move to `third_party`?

Thanks for the comment @jwhui.  Can you provide a little more info on what ot-efr32 files require the OpenThread license?  Is it the policy that everything that is not in third_party should have the OpenThread license?

Reply:
Yes, correct.